soco/data_structures.py

```diff
@@ -504,21 +504,20 @@ def from_element(cls, element):     # pylint: disable=R0914
 
         # CAUTION: This implementation deviates from the spec.
         # Elements are normally required to have a `restricted` tag, but
-        # Spotify direct violates this. To make it work, a missing restricted
+        # Spotify Direct violates this. To make it work, a missing restricted
         # tag is interpreted as `restricted = True`.
         restricted = element.get('restricted', None)
         restricted = False if restricted in [0, 'false', 'False'] else True
 
-        # There must be a title. According to spec, it should be the first
-        # child, but Sonos does not abide by this
+        # Similarily, all elements should have a title tag, but Spotify Direct
+        # does not comply
         title_elt = element.find(ns_tag('dc', 'title'))
         if title_elt is None:
-            raise DIDLMetadataError(
-                "Missing title element")
-        title = title_elt.text
-        if title_elt.text is None:
+            title = None
```

Comment:
The one bit I don't like about this PR is line 516. According to the docs for the DIDL-Lite elements, the title is mandatory and it says nothing about a possible None value, so I think it should default to the empty string in this case as well.

Reply:
I intentionally set title to None, because in this case, the title really nonexistent, not empty. Defaulting to an empty string would make it impossible to distinguish between missing and empty title.
But if you want the empty string as a default (might integrate better with other code?), that's also fine with me.

Reply:
Yeah, I got that, and that might make sense. My point is just that the docs for the DidlLite Element says that the title is mandatory and a string. So either we should default to a string, or, change the DidlLiteElement docstring away from spec, to document the possibility of a None value for the title.

Initially I was against the idea of changing DidlLite implementation too much away from spec, but since we are already down that rabbit hole, I am now thinking that it might actually be the better option.

So if we go with that option, I would ask you to:

 1. Change the docstring in the appropriate DidlLite class
 2. Highlight in the release notes thread, that this fix introduced a change of possibly data types for a core data type